ffs/types.go

```diff
@@ -109,6 +110,7 @@ type StorageJob struct {
 	Cid        cid.Cid
 	Status     JobStatus
 	ErrCause   string
+	DealInfo   []deals.StorageDealInfo
```

Comment:
We now store storage deal state in `StorageJob` in the `sjstore`. We were previously already storing storage job error data there, this extends that to also store the rest of the deal information.

Reply:
Cool. Important to tell people that this is Job related deals, not complete photo of things which is CidInfo, right?

e.g: 
- Cid1 is first executed by Job1 making 1 deal.
- Cid1 pushes a new config with rep factor 2. Job2 executes that config and creates 1 extra deal.

Result:
- Job1.DealInfo has 1 dea..
- Job2.DealInfo has 1 deals, since this job only created 1 deal.
- CidInfo should have the 2 deals information.

Same page?

Reply:
Yea same page. I want to be really clear about the difference between jobs and storage state. In an upcoming PR, I'd like to rename `CidInfo` to `StorageInfo` for that reason. So then we have `StorageJob`s and `StorageInfo`